apps/system/js/sync_manager.js

```diff
@@ -139,10 +140,20 @@
                 Service.request('SyncStateMachine:enable').then(resolve);
                 break;
               case 'enabling':
-              case 'errored':
                 this.updateState('disabled');
                 resolve();
                 break;
+              case 'errored':
+                if (this.error != ERROR_UNVERIFIED_ACCOUNT) {
+                  this.updateState('disabled');
+                  resolve();
+                  break;
+                }
+                // We try to enable Sync. This will check if the account is
+                // verified or not. If it's not verified, we go back to the
+                // errored state, otherwise, the user logs in.
+                Service.request('SyncStateMachine:enable').then(resolve);
```

Comment:
nit: maybe have the `if` be the special case (`this.error == ERROR_UNVERIFIED_ACCOUNT`) and let the general case below. a tiny bit more readable.